src/herbpy/herbrobot.py

```diff
@@ -271,7 +270,72 @@ def SetStiffness(self, stiffness):
         self.head.SetStiffness(stiffness)
         self.left_arm.SetStiffness(stiffness)
         self.right_arm.SetStiffness(stiffness)
-
+        
+    def DetectHuman(self, env, orhuman=2, load_hum=True, 
+                    hrc=False, continuous=True,      
+                    herb_sim=True, enable_legs=True,             
+                    segway_sim=True, action='stamp'):
+        """Use the kinbody detector to detect objects and add
+        them to the environment
+        """
+        import rospy
+        from tf import TransformListener, transformations
+
+        node_name = 'detecthuman'  
+
+        if orhuman==0: 
+            if load_hum==True:
+                #kin 1 - load a skeleton
+                import or_skeletons.load_skeletons as sk
+                humans = []
+                logger.info('Humans_tracking')
+        elif orhuman==1:
+            if load_hum==True:
+                #kin 1- load the human from openrave
+                import humanpy.hum_kin1 as sk
+                humans = []
+                logger.info('Humans_tracking')
+        elif orhuman==2:
+            if load_hum==True:
+                #kin 2 - load the human from openrave
+                import humanpy.hum_kin2 as sk
+                humans = []
+                logger.info('Humans_tracking')
+            if hrc==True:
+                from hrc import assistance                 
+                humans_hrc = []
+                logger.info('HRC')
+                
+        try:
+            rospy.init_node(node_name, anonymous=True)
+        except rospy.exceptions.ROSException:
+            pass
```

Comment:
We can only call `init_node` once per process, so we always do so in the main script where we call `herbpy.initialize` to prevent multiple modules from conflicting with each other. Please change this function to throw an exception if a ROS node is not already running, instead of starting one yourself.